src/Controller/MessageController.php

```diff
@@ -23,35 +24,45 @@ class MessageController extends AbstractController
     /**
      * TODO: cover this method with tests, and refactor the code (including other files that need to be refactored)
      */
-    #[Route('/messages')]
+    #[Route('/messages', name: 'app_message_list', methods: ['GET'])]
     public function list(Request $request, MessageRepository $messages): Response
     {
-        $messages = $messages->by($request);
-  
```

Comment:
Request object must never be used by services. Otherwise, same methods cannot be later reused in messenger where there is no request.